### PR TITLE
Cal 262 refactor edit teams according to the design reference

### PR DESCRIPTION
**Pull request by alishaz-polymath**
- Changed settings/teams look and flow according to the design reference 
- Changed settings/teams-edit flow according to the design reference
- Added logo, bio, branding columns to team table
- attached ImageUploader to the team logo

What's left:

- bio, logo, branding, slug patch API
- Disband team directly from the dropdown menu in settings/teams page
- Pull latest data without refresh after an action in settings/teams page
- Pull latest data without refresh after an action in settings/teams-edit page
- Code clean-up

### CODE REVIEW

**Comment by alishaz-polymath** [comment]
What's left:

- slug conflict check at Patch time
- Disband team directly from the dropdown menu in settings/teams page
- Pull latest data without refresh after an action in settings/teams page
- Pull latest data without refresh after an action in settings/teams-edit page
- Code clean-up

**Comment by alishaz-polymath** [comment]
What's left:

- slug conflict check at Patch time
- Pull latest data without refresh after an action in settings/teams page
- Pull latest data without refresh after an action in settings/teams-edit page
- Code clean-up

**Comment by alishaz-polymath** [comment]
What's left:

- slug conflict check at Patch time
- Code clean-up

**Review by emrysal — CHANGES_REQUESTED** [review]
Note: I did not do a end-to-end review, so code-review only (there's a few changes so I think it makes sense to do that after the initial changes are made).

Ps. `Dialog` has a much neater interface than `Modal` so you may want to consider where-ever possible.

**Review by KATT — CHANGES_REQUESTED** [review]

File: components/team/TeamListItem.tsx
Comment (alishaz-polymath) [reply components/team/TeamListItem.tsx]: Will get this sorted in a new PR where I set up BASE_URL in publicRuntimeConfig inside next.config.js, as we discussed.

File: pages/api/teams/[team]/profile.ts
Comment (KATT) [reply pages/api/teams/[team]/profile.ts:37]: +1

Comment (KATT) [reply pages/api/teams/[team]/profile.ts:37]: I see it's added - you only need this if you want to give a nice error message and it should be changed to `.findUnique`

File: components/team/EditTeam.tsx
Comment (KATT) [inline components/team/EditTeam.tsx:23]: a form lib like https://react-hook-form.com/ would get rid of this

File: components/team/EditTeam.tsx
Comment (KATT) [inline components/team/EditTeam.tsx:27]: I personally prefer using search params when modals are open. Subjective opinion but I like when things can be linked

https://github.com/calendso/calendso/blob/8e9703545af208aa073783a9f0643e635943e035/pages/event-types/index.tsx#L46
https://github.com/calendso/calendso/blob/8e9703545af208aa073783a9f0643e635943e035/pages/event-types/index.tsx#L88-L93

File: components/team/EditTeam.tsx
Comment (KATT) [inline components/team/EditTeam.tsx:54]: `useMutation` from react-query

File: components/team/TeamListItem.tsx
Comment (KATT) [inline components/team/TeamListItem.tsx:76]: Use `<Button />`

**Review by KATT — APPROVED** [review]
- Looks ok to me - overall follows the coding structure we could see in the app before react-query.
- I have **not** tested the actual functionality
- It would be greatly simplified with react-query and react-hook-form
- Would benefit from making some reusable components in layouting

File: components/team/TeamListItem.tsx
Comment (alishaz-polymath) [reply components/team/TeamListItem.tsx:76]: @Jaibles is currently setting up a new design for this, hence I'm just waiting for him to finish that. Once that is done, I'll modify this bit accordingly and update the use of `<button>` or similar elements into the components we're using elsewhere.

**Comment by alishaz-polymath** [comment]
@KATT Do you suggest that I work on utilizing the react-query and react-hook-form within this PR itself, or should I open it 'after' this PR is merged?
I am fine either way, but I feel we may even start going through each page (on a separate PR) for revamping them to utilize the latest standards such as the react-query and react-hook-form, for example.

**Comment by KATT** [comment]
> @KATT Do you suggest that I work on utilizing the react-query and react-hook-form within this PR itself, or should I open it 'after' this PR is merged?

It's above average in coding style compared to the full codebase + it has new features.  - I'd say test it and ship it and do a follow-up refactor if you want. :)

But it's not up to me - get sign-off from @PeerRich / @baileypumfleet

**Comment by alishaz-polymath** [comment]
Will do. Currently waiting on @emrysal to review the changes and see if any further changes are required according to him. Thanks a lot @KATT . Really appreciate you reviewing it.

**Comment by PeerRich** [comment]
i recommend shipping this PR and then improvements in a new PR

**Comment by KATT** [comment]
@alishaz-polymath if you push this to the calendso origin (instead of your fork) and open a new PR you'll get vercel/linting/etc working

**Comment by alishaz-polymath** [comment]
> 
> 
> @alishaz-polymath if you push this to the calendso origin (instead of your fork) and open a new PR you'll get vercel/linting/etc working

Do you mean if I create a new branch in calendso/calendso and then push these changes to that branch and open a new PR?

**Comment by KATT** [comment]
> > 
> > 
> > @alishaz-polymath if you push this to the calendso origin (instead of your fork) and open a new PR you'll get vercel/linting/etc working
> 
> Do you mean if I create a new branch in calendso/calendso and then push these changes to that branch and open a new PR?

Yup, that ☝️

**Comment by alishaz-polymath** [comment]
> 
> 
> > > @alishaz-polymath if you push this to the calendso origin (instead of your fork) and open a new PR you'll get vercel/linting/etc working
> > 
> > 
> > Do you mean if I create a new branch in calendso/calendso and then push these changes to that branch and open a new PR?
> 
> Yup, that ☝️

Do you want me to do that now (on this PR itself) or from now on, for the new PRs (excluding this one)?

**Comment by KATT** [comment]
> > 
> > 
> > > > @alishaz-polymath if you push this to the calendso origin (instead of your fork) and open a new PR you'll get vercel/linting/etc working
> > > 
> > > 
> > > Do you mean if I create a new branch in calendso/calendso and then push these changes to that branch and open a new PR?
> > 
> > Yup, that ☝️
> 
> Cool! Do you want me to do that now (on this PR itself) or from now on, for the new PRs (excluding this one)?

@alishaz-polymath would be great if you do it for this one too so the tests are run! 🙏

**Comment by emrysal** [comment]
Minor issue with the background colour set to `bg-white` here:

image

This should go to `/team/abcd`  - The /team prefix is an architectural constraint we are agreed to live with - allows for findUnique either on username (with /abcd) or team slug (/team/abcd). Comment out for now so this can be merged without depending on the team profiles.

image

Class `space-x-2` to be added to button wrapper.

image

I can open multiple "Three dots" simultaneously - similarly on the Team listing.

image
image

This does not seem to do anything?
image


There is some differences to the design which I think should be as per-design:

* Member should be Members in the member list heading.
* Branding heading is not part of the design, as it doesn't need to be there best to remove.
* "Upload a logo" has changed to "Change logo" - @Jaibles which do you prefer?
* The about section has a text underneath the about, and no placeholder. IMO it can be both in the placeholder as underneath the About textarea, but the full version "A few sentences about your team. This will appear on your team's URL page." is nicer due to the added explanation of where it is used.

The field "Username" should become "My personal URL" as per design, but I think the design isn't right either here - how about "My team URL"?

FUTURE IMPROVEMENT (ignore for now):
When pressing "Teams" in the header whilst on a team profile page, the user should be redirected to the Team index in my opinion.

P.s. Invite on staging got a 500 error - but as my invite was created I think this is due to emails not working on staging @baileypumfleet. Can confirm I did not receive the invite email.

**Comment by alishaz-polymath** [comment]
> Minor issue with the background colour set to `bg-white` here:
> 
> image
> 
> This should go to `/team/abcd` - The /team prefix is an architectural constraint we are agreed to live with - allows for findUnique either on username (with /abcd) or team slug (/team/abcd). Comment out for now so this can be merged without depending on the team profiles.
> 
> image
> 
> Class `space-x-2` to be added to button wrapper.
> 
> image
> 
> I can open multiple "Three dots" simultaneously - similarly on the Team listing.
> 
> image
> image
> 
> This does not seem to do anything?
> image
> 
> There is some differences to the design which I think should be as per-design:
> 
> * Member should be Members in the member list heading.
> * Branding heading is not part of the design, as it doesn't need to be there best to remove.
> * "Upload a logo" has changed to "Change logo" - @Jaibles which do you prefer?
> * The about section has a text underneath the about, and no placeholder. IMO it can be both in the placeholder as underneath the About textarea, but the full version "A few sentences about your team. This will appear on your team's URL page." is nicer due to the added explanation of where it is used.
> 
> The field "Username" should become "My personal URL" as per design, but I think the design isn't right either here - how about "My team URL"?
> 
> FUTURE IMPROVEMENT (ignore for now):
> When pressing "Teams" in the header whilst on a team profile page, the user should be redirected to the Team index in my opinion.
> 
> P.s. Invite on staging got a 500 error - but as my invite was created I think this is due to emails not working on staging @baileypumfleet. Can confirm I did not receive the invite email.

--The dropdown menu (multiple openings plus not closing when clicked outside the menu) will be taken care of in a separate PR which will add Radix-ui dropdown and replace this entirely.

Will make the other changes and update!!

**Review by emrysal — APPROVED** [review]
Approved when you've resolved the merge conflicts 👍

**Comment by alishaz-polymath** [comment]
> Approved when you've resolved the merge conflicts 👍

On it!

**Pull request merged**
